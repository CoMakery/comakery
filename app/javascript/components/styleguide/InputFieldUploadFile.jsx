import React from 'react'
import classNames from 'classnames'
import InputField from './InputField'

class InputFieldUploadFile extends React.Component {
  render() {
    const {
      className,
      ...other
    } = this.props

    const classnames = classNames(
      'input-field__upload-file',
      className
    )

    return (
      <React.Fragment>
        <InputField
          className={classnames}
          type="file"
          symbolLimit={0}
          {...other}
        />
      </React.Fragment>
    )
  }
}

export default InputFieldUploadFile
