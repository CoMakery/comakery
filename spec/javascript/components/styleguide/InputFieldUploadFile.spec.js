import React from 'react'
import { shallow, mount, render } from 'enzyme'
import InputFieldUploadFile from 'components/styleguide/InputFieldUploadFile'

describe('InputFieldUploadFile', () => {
  it('renders correctly without props', () => {
    const wrapper = shallow(<InputFieldUploadFile />)

    expect(wrapper).toMatchSnapshot()
    expect(wrapper.find('.input-field__upload-file').props().type).toBe('file')
    expect(wrapper.exists('.input-field--title--counter')).not.toBe()
  })

  it('renders correctly with custom className', () => {
    const wrapper = shallow(<InputFieldUploadFile className="__test" />)

    expect(wrapper.exists('.input-field__upload-file.__test')).toBe(true)
  })
})
