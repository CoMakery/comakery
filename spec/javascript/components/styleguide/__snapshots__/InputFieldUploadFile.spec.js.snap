// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputFieldUploadFile renders correctly without props 1`] = `
<Fragment>
  <InputField
    checkboxText=""
    checked={false}
    className="input-field__upload-file"
    copyOnClick={false}
    disabled={false}
    errorText=""
    eventHandler={[Function]}
    imgInputRef={null}
    imgPreviewDimensions="40x40"
    imgPreviewUrl=""
    imgRequirements=""
    imgVerifier={[Function]}
    max=""
    min=""
    name="field"
    pattern=".*"
    placeholder="Please enter value"
    readOnly={false}
    recommended={false}
    required={false}
    selectEntries={Array []}
    step=""
    symbolLimit={0}
    title="title"
    type="file"
    value=""
  />
</Fragment>
`;
